src/cgame/cg_weapons.cpp

```diff
@@ -2207,9 +2207,34 @@ void CG_NextWeapon_f()
 		return;
 	}
 
-	// HACK: This only works with two weapons.
-	// See: https://github.com/Unvanquished/Unvanquished/issues/544#issuecomment-619299752
-	trap_SendClientCommand( "itemtoggle blaster\n" );
+	// Find next weapon in inventory.
+	weapon_t nextWeapon = WP_NONE;
```

Comment:
This doesn't really work in the general case of more than two weapons. So if you don't want to spend time fixing it, I'd rather keep `itemtoggle blaster` which is simpler.

Reply:
Why? It is expected to remember the first weapon it sees before finding the current weapon (to handle the case of the current weapon being the last one) until it finds the current weapon and, if it's not the end of the list, remember the next weapon and stop.

Reply:
Doing what you describe would work, but what the code does is always return the 2nd other weapon. You'd need to add some condition like `w > cg.snap->ps.weapon`.

Reply:
hmmm

Reply:
I rewrote it entirely, also made a reusable function, can you proof read that again?